Duplicati/Library/Main/Operation/Backup/BackendUploader.cs

```diff
@@ -318,7 +318,15 @@ private async Task<bool> DoWithRetry(Func<Task> method, FileEntryItem item, Work
             for (retryCount = 0; retryCount <= m_options.NumberOfRetries; retryCount++)
             {
                 if (m_options.RetryDelay.Ticks != 0 && retryCount != 0)
-                    await Task.Delay(m_options.RetryDelay).ConfigureAwait(false);
+                {
+                    TimeSpan delay;
+                    if (m_options.RetryDelayExponential)
+                        delay = TimeSpan.FromTicks(m_options.RetryDelay.Ticks * (2L << Math.Min(63, retryCount - 1)));

```

Comment:
Was there a reason to prefer the form
```
x * (2L << (y - 1))
```
instead of just
```
x << y
```

Reply:
I'm taking `(y - 1)` as opposed to `y` because `y` (i.e. the retries counter) is first incremented after the first failure. So, taking `y` would result in the first delay being double of what the user has set with the `retry-delay` argument, which I don't think is the behaviour the user would expect.

As for using the form `x * (2L << (y-1))` instead of `x << (y - 1)`, I have now changed it to `x << (y -1)`, which is indeed more concise and means the same thing.